### ISSUE TITLE
/v1/auth doesn't accept a request with no blockchain id

### ISSUE THREAD

**Issue by larrysalibra** [blockstack/blockstack-core#483]
This prevents users from logging into apps before registering a name. see https://github.com/blockstack/blockstack.js/issues/230#issue-241190457

```
127.0.0.1 - - [07/Jul/2017 16:29:50] "POST /v1/node/drivers/storage/dropbox?index=1 HTTP/1.1" 200 -
[2017-07-07 16:29:51,042] [DEBUG] [rpc:3457] (29596.123145509597184) 
full path: /v1/auth?authRequest=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJ2ZXJzaW9uIjoxLCJibG9ja2NoYWluX2lkIjpudWxsLCJhcHBfcHJpdmF0ZV9rZXkiOiIxNDYwYWIyY2RjZmE1NDQwNzc5YWYwZDA0NWIzZTFlMjE5MjY4OGRjZTA5NDk4YWMyNDBkMTdjNzA2YWRiOThkIiwiYXBwX2RvbWFpbiI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMCIsIm1ldGhvZHMiOlsiIl0sImFwcF9wdWJsaWNfa2V5cyI6W3sicHVibGljX2tleSI6IjAzYTJhZmYyODhlYjI1NzVjZjE3ZTBjODc0NDZlNWExMDdmOTFkZjMzMjk5MjNkNDNmMDhmYTFmNzdlZDE0MTNmMCIsImRldmljZV9pZCI6IjAifV0sImRldmljZV9pZCI6IjAifQ.-uT-lOrvQDBZJWdg8p53LmEYBw1C8dVyGSAn96nR49MGSlNXP0vD7JsasjI6cbn9JSqGPFq1EpPLaHACkmyMcQ
method: GET
path: /v1/auth
qs: {'authRequest': 'eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJ2ZXJzaW9uIjoxLCJibG9ja2NoYWluX2lkIjpudWxsLCJhcHBfcHJpdmF0ZV9rZXkiOiIxNDYwYWIyY2RjZmE1NDQwNzc5YWYwZDA0NWIzZTFlMjE5MjY4OGRjZTA5NDk4YWMyNDBkMTdjNzA2YWRiOThkIiwiYXBwX2RvbWFpbiI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMCIsIm1ldGhvZHMiOlsiIl0sImFwcF9wdWJsaWNfa2V5cyI6W3sicHVibGljX2tleSI6IjAzYTJhZmYyODhlYjI1NzVjZjE3ZTBjODc0NDZlNWExMDdmOTFkZjMzMjk5MjNkNDNmMDhmYTFmNzdlZDE0MTNmMCIsImRldmljZV9pZCI6IjAifV0sImRldmljZV9pZCI6IjAifQ.-uT-lOrvQDBZJWdg8p53LmEYBw1C8dVyGSAn96nR49MGSlNXP0vD7JsasjI6cbn9JSqGPFq1EpPLaHACkmyMcQ'}
headers:
 origin: http://localhost:3000
accept-language: en-US,en;q=0.8
accept-encoding: gzip, deflate, br
connection: keep-alive
accept: */*
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.115 Safari/537.36
host: localhost:6270
referer: http://localhost:3000/auth?authRequest=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJqdGkiOiI1NTY4ZDUwYy0wOTUwLTQ1MWEtYWE2NC05NzBmOTFhNWJjMGQiLCJpYXQiOjE0OTk0MTQzMzEsImV4cCI6MTQ5OTQxNzkzMSwiaXNzIjoiZGlkOmJ0Yy1hZGRyOjFQaVZFc3dyU0hqS1Z5WGRHM1BDYjl0TDNIdXNYRVJHOVMiLCJwdWJsaWNfa2V5cyI6WyIwMmM4ZjllODU4NDk2OGNlYWNhYTFmNmE3ODY4ZjQxYWYzMjA2YjI1ODQyMDIzZjAyYWY5YjMzZjY3ZTcwNTBlM2MiXSwiZG9tYWluX25hbWUiOiJodHRwOi8vbG9jYWxob3N0OjUwMDAiLCJtYW5pZmVzdF91cmkiOiJodHRwOi8vbG9jYWxob3N0OjUwMDAvbWFuaWZlc3QuanNvbiIsInJlZGlyZWN0X3VyaSI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMC8iLCJzY29wZXMiOlsiIl19.G15kf95mPm4FoBJT6AYE7lEIZPHJqL7Fdd_tQr7ddQXV2P1vjm03uysXIW2DcMffJkRZ_bYFkJOY6npTxerY9g
authorization: bearer blockstack_integration_test_api_password

[2017-07-07 16:29:51,043] [DEBUG] [app:113] (29596.123145509597184) Not a valid token
[2017-07-07 16:29:51,043] [DEBUG] [rpc:3479] (29596.123145509597184) Authenticated with password
[2017-07-07 16:29:51,047] [DEBUG] [rpc:442] (29596.123145509597184) Authentication request is not current; trying legacy
[2017-07-07 16:29:51,048] [ERROR] [rpc:448] (29596.123145509597184) 'app_public_key' is a required property

Failed validating 'required' in schema:
    {'properties': {'app_domain': {'anyOf': [{'pattern': '^([a-z0-9+]+)://([a-zA-Z0-9\\-_.~%#?&\\\\:/=]+)$',
                                              'type': 'string'},
                                             {'pattern': '^([a-zA-Z0-9\\-_.~%#?&\\\\:/=]+)$',
                                              'type': 'string'}]},
                    'app_public_key': {'pattern': '^([0-9a-fA-F]+)$',
                                       'type': 'string'},
                    'methods': {'items': {'pattern': '^[a-zA-Z_][a-zA-Z0-9_.]+$',
                                          'type': 'string'},
                                'type': 'array'}},
     'required': ['app_domain', 'app_public_key', 'methods'],
     'type': 'object'}

On instance:
    {u'app_domain': u'http://localhost:5000',
     u'app_private_key': u'1460ab2cdcfa5440779af0d045b3e1e2192688dce09498ac240d17c706adb98d',
     u'app_public_keys': [{u'device_id': u'0',
                           u'public_key': u'03a2aff288eb2575cf17e0c87446e5a107f91df3329923d43f08fa1f77ed1413f0'}],
     u'blockchain_id': None,
     u'device_id': u'0',
     u'methods': [u''],
     u'version': 1}
Traceback (most recent call last):
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/blockstack-0.14.2.0-py2.7.egg/blockstack_client/rpc.py", line 443, in GET_auth
    jsonschema.validate(decoded_token['payload'], APP_SESSION_REQUEST_SCHEMA_OLD )
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/jsonschema/validators.py", line 541, in validate
    cls(schema, *args, **kwargs).validate(instance)
  File "/Users/larry/git/blockstack-testing/blockstack-venv/lib/python2.7/site-packages/jsonschema/validators.py", line 130, in validate
    raise error
ValidationError: 'app_public_key' is a required property

Failed validating 'required' in schema:
    {'properties': {'app_domain': {'anyOf': [{'pattern': '^([a-z0-9+]+)://([a-zA-Z0-9\\-_.~%#?&\\\\:/=]+)$',
                                              'type': 'string'},
                                             {'pattern': '^([a-zA-Z0-9\\-_.~%#?&\\\\:/=]+)$',
                                              'type': 'string'}]},
                    'app_public_key': {'pattern': '^([0-9a-fA-F]+)$',
                                       'type': 'string'},
                    'methods': {'items': {'pattern': '^[a-zA-Z_][a-zA-Z0-9_.]+$',
                                          'type': 'string'},
                                'type': 'array'}},
     'required': ['app_domain', 'app_public_key', 'methods'],
     'type': 'object'}

On instance:
    {u'app_domain': u'http://localhost:5000',
     u'app_private_key': u'1460ab2cdcfa5440779af0d045b3e1e2192688dce09498ac240d17c706adb98d',
     u'app_public_keys': [{u'device_id': u'0',
                           u'public_key': u'03a2aff288eb2575cf17e0c87446e5a107f91df3329923d43f08fa1f77ed1413f0'}],
     u'blockchain_id': None,
     u'device_id': u'0',
     u'methods': [u''],
     u'version': 1}
[2017-07-07 16:29:51,060] [DEBUG] [rpc:450] (29596.123145509597184) Invalid decoded token: {u'app_private_key': u'1460ab2cdcfa5440779af0d045b3e1e2192688dce09498ac240d17c706adb98d', u'blockchain_id': None, u'methods': [u''], u'app_domain': u'http://localhost:5000', u'app_public_keys': [{u'public_key': u'03a2aff288eb2575cf17e0c87446e5a107f91df3329923d43f08fa1f77ed1413f0', u'device_id': u'0'}], u'version': 1, u'device_id': u'0'}
[2017-07-07 16:29:51,060] [DEBUG] [rpc:452] (29596.123145509597184) Invalid token
127.0.0.1 - - [07/Jul/2017 16:29:51] "GET /v1/auth?authRequest=eyJ0eXAiOiJKV1QiLCJhbGciOiJFUzI1NksifQ.eyJ2ZXJzaW9uIjoxLCJibG9ja2NoYWluX2lkIjpudWxsLCJhcHBfcHJpdmF0ZV9rZXkiOiIxNDYwYWIyY2RjZmE1NDQwNzc5YWYwZDA0NWIzZTFlMjE5MjY4OGRjZTA5NDk4YWMyNDBkMTdjNzA2YWRiOThkIiwiYXBwX2RvbWFpbiI6Imh0dHA6Ly9sb2NhbGhvc3Q6NTAwMCIsIm1ldGhvZHMiOlsiIl0sImFwcF9wdWJsaWNfa2V5cyI6W3sicHVibGljX2tleSI6IjAzYTJhZmYyODhlYjI1NzVjZjE3ZTBjODc0NDZlNWExMDdmOTFkZjMzMjk5MjNkNDNmMDhmYTFmNzdlZDE0MTNmMCIsImRldmljZV9pZCI6IjAifV0sImRldmljZV9pZCI6IjAifQ.-uT-lOrvQDBZJWdg8p53LmEYBw1C8dVyGSAn96nR49MGSlNXP0vD7JsasjI6cbn9JSqGPFq1EpPLaHACkmyMcQ HTTP/1.1" 401 -
[2017-07-07 16:29:51,142] [DEBUG] [registrar:825] (29596.123145505390592) register all pending preorders in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:51,143] [DEBUG] [registrar:842] (29596.123145505390592) put zonefile hashes for registered names in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:51,144] [DEBUG] [registrar:859] (29596.123145505390592) replicate all pending zone files and profiles for updates /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:51,145] [DEBUG] [registrar:876] (29596.123145505390592) transfer all names in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:51,148] [DEBUG] [config:877] (29596.123145505390592) Change blockstack-client.client_version to 0.14.2.0
[2017-07-07 16:29:51,154] [DEBUG] [registrar:894] (29596.123145505390592) replicate all pending zone files for name imports in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:51,154] [DEBUG] [registrar:911] (29596.123145505390592) clean out other confirmed operations
[2017-07-07 16:29:51,155] [DEBUG] [registrar:934] (29596.123145505390592) Sleep for 1
[2017-07-07 16:29:52,156] [DEBUG] [registrar:825] (29596.123145505390592) register all pending preorders in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:52,158] [DEBUG] [registrar:842] (29596.123145505390592) put zonefile hashes for registered names in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:52,158] [DEBUG] [registrar:859] (29596.123145505390592) replicate all pending zone files and profiles for updates /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:52,159] [DEBUG] [registrar:876] (29596.123145505390592) transfer all names in /tmp/blockstack-run-scenario.blockstack_integration_tests.scenarios.portal_test_env/client/queues.db
[2017-07-07 16:29:52,161] [DEBUG] [config:877] (29596.123145505390592) Change blockstack-client.client_version to 0.14.2.0
```

**Comment by kantai** [comment]
The problem here is that the request submitted uses `'app_public_keys'`, which is different from the required `'app_public_key'`

**Comment by jcnelson** [comment]
It should now.

**Comment by kantai** [comment]
Okay -- I just pushed a unit test for this auth string.

This auth string *also* passed a methods property with an empty string method `[u'']`, so the schema now supports that, and then core will filter it out to handle it a little bit more gracefully. Now this auth string should be accepted by core (on the head of `rc-0.14.3`)

**Comment by kantai** [comment]
Closing as this has been addressed in rc-0.14.3
